eval/src/tests/instruction/mixed_simple_join_function/mixed_simple_join_function_test.cpp

```diff
@@ -144,108 +161,85 @@ struct LhsRhs {
-                                    LhsRhs("y5", "x3y5", 5, 15, Overlap::INNER),
-                                    LhsRhs("y5", "y5z3", 5, 15, Overlap::OUTER),
-                                    LhsRhs("x3y5", "y5", 15, 5, Overlap::INNER),
-                                    LhsRhs("y5z3", "y5", 15, 5, Overlap::OUTER)})
-                        {
-                            vespalib::string left = adjust_param(params.lhs, left_float, left_mut, false);
-                            vespalib::string right = adjust_param(params.rhs, right_float, right_mut, true);
-                            vespalib::string expr = fmt(op_pattern, left.c_str(), right.c_str());
-                            TEST_STATE(expr.c_str());
-                            Primary primary = Primary::RHS;
-                            if (params.overlap == Overlap::FULL) {
-                                bool w_lhs = ((left_float == float_result) && left_mut);
-                                bool w_rhs = ((right_float == float_result) && right_mut);
-                                if (w_lhs && !w_rhs) {
-                                    primary = Primary::LHS;
-                                }
-                            } else if (params.lhs_size > params.rhs_size) {
-                                primary = Primary::LHS;
-                            }
-                            bool pri_mut = (primary == Primary::LHS) ? left_mut : right_mut;
-                            bool pri_float = (primary == Primary::LHS) ? left_float : right_float;
-                            int p_inplace = -1;
-                            if (pri_mut && (pri_float == float_result)) {
-                                p_inplace = (primary == Primary::LHS) ? 0 : 1;
-                            }
-                            verify_optimized(expr, primary, params.overlap, pri_mut, params.factor, p_inplace);
-                        }
-                    }
+    for (bool left_mut: {false, true}) {
+        for (bool right_mut: {false, true}) {
+            for (const char *op_pattern: {"%s+%s", "%s-%s", "%s*%s"}) {
+                for (const LhsRhs &params:
+                         {       LhsRhs("y5",   "y5", 5,  5, Overlap::FULL),
+                                 LhsRhs("y5", "x3y5", 5, 15, Overlap::INNER),
+                                 LhsRhs("y5", "y5z3", 5, 15, Overlap::OUTER),
+                                 LhsRhs("x3y5", "y5", 15, 5, Overlap::INNER),
+                                 LhsRhs("y5z3", "y5", 15, 5, Overlap::OUTER)})
+                {
+                    vespalib::string left = adjust_param(params.lhs, left_mut, false);
+                    vespalib::string right = adjust_param(params.rhs, right_mut, true);
+                    vespalib::string expr = fmt(op_pattern, left.c_str(), right.c_str());
+                    TEST_STATE(expr.c_str());
+                    verify_optimized(expr, params.overlap, params.factor,
```

Comment:
it might not be a good match to use the EvalFixture::verify function for this, since we already map out various combinations for the parameters and would benefit from having knowledge of the cell types up front for each case. We might also want to use a fixed expression with 'a' and 'b' and only modify the input parameters as well...